openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -585,394 +578,403 @@ OpenAssessment.ResponseView.prototype = {
-                        'Only the supported file types can be uploaded. ' +
-                        'If you have questions, please reach out to the course team.'
-                    )
-                );
-                errorCheckerTriggered = true;
-                break;
-            }
+     */
+    prepareUpload(files, uploadType, descriptions) {
+      this.files = null;
+      this.filesType = uploadType;
+      this.filesUploaded = false;
+
+      let errorCheckerTriggered = false;
+
+      for (let i = 0; i < files.length; i++) {
+        if (files[i].size > this.MAX_FILE_SIZE) {
+          this.baseView.toggleActionError(
+            'upload',
+            gettext(
+              'Individual file size must be {max_files_mb}MB or less.',
+            ).replace(
+              '{max_files_mb}',
+              this.MAX_FILES_MB,
+            ),
+          );
+          errorCheckerTriggered = true;
+          break;
         }
 
-        if (this.getSavedFileCount(false) + files.length > this.data.MAXIMUM_FILE_UPLOAD_COUNT ) {
-            var msg = gettext('Only ' + this.data.MAXIMUM_FILE_UPLOAD_COUNT + ' files can be saved.');
-            this.baseView.toggleActionError(
-                'upload',
-                gettext(msg)
-            );
-            errorCheckerTriggered = true;
+        if (!this.isUploadSupported(files[i], uploadType)) {
+          this.baseView.toggleActionError(
+            'upload',
+            gettext(
+              'File upload failed: unsupported file type. '
+              + 'Only the supported file types can be uploaded. '
+              + 'If you have questions, please reach out to the course team.',
+            ),
+          );
+          errorCheckerTriggered = true;
+          break;
         }
+      }
 
-        if (!errorCheckerTriggered) {
-            this.baseView.toggleActionError('upload', null);
-            if (files.length > 0) {
-                this.files = files;
-            }
-            this.updateFilesDescriptionsFields(files, descriptions, uploadType);
-        }
+      if (this.getSavedFileCount(false) + files.length > this.data.MAXIMUM_FILE_UPLOAD_COUNT) {
+        const msg = gettext(`Only ${this.data.MAXIMUM_FILE_UPLOAD_COUNT} files can be saved.`);
+        this.baseView.toggleActionError(
+          'upload',
+          gettext(msg),
+        );
+        errorCheckerTriggered = true;
+      }
 
-        if (this.files === null) {
-            $(this.element).find('.file__upload').prop('disabled', true);
+      if (!errorCheckerTriggered) {
+        this.baseView.toggleActionError('upload', null);
+        if (files.length > 0) {
+          this.files = files;
         }
-    },
-
-    /**
+        this.updateFilesDescriptionsFields(files, descriptions, uploadType);
+      }
+
+      if (this.files === null) {
+        $(this.element).find('.file__upload').prop('disabled', true);
+      }
+    }
+
+   isUploadSupported = (file, uploadType) => {
```

Comment:
Is there a particular reason to move this to the bottom?

Reply:
moving from a local function to a class method.